File: FinalProject/app/src/main/java/com/example/finalproject/Item.java
package com.example.finalproject;

public class Item {
    private int cost, rarity, category;
    private String name;

    public Item(){}

    public Item(String name){
        this.name=name;
    }

    public int getCost() {
        return cost;
    }

    public void setCost(int cost) {
        this.cost = cost;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getRarity() {
        return rarity;
    }

    public void setRarity(int rarity) {
        this.rarity = rarity;
    }

    public int getCategory() {
        return category;
    }

    public void setCategory(int category) {
        this.category = category;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/service/classes/items/Item.kt
package com.example.finalproject.service.classes.items

import com.example.finalproject.service.classes.Unit

open class Item(
    name: String,
    id: Int,
    val costSell: Int,
    val costBuy: Int,
    val rarity: Int,
    val category: Int
) : Unit(name = name, id = id) {

    constructor() : this("", -1, 0, 0, 0, -1)

    constructor(item: Item) : this(
        item.name,
        item.id,
        item.costSell,
        item.costBuy,
        item.rarity,
        item.category
    )
}